narwhals/utils.py

```diff
@@ -194,8 +194,8 @@ def validate_laziness(items: Iterable[Any]) -> None:
         all(isinstance(item, LazyFrame) for item in items)
     ):
         return
-    msg = "The items to concatenate should either all be eager, or all lazy"
-    raise NotImplementedError(msg)
+    msg = f"The items to concatenate should either all be eager, or all lazy, got: {[type(item) for item in items]}"
+    raise TypeError(msg)
```

Comment:
@DeaMariaLeon it's probably always a good practice to show what we got (vs what we expected) in error messages - no blame of course, just a suggestion for the future! 🤗

Reply:
Thanks a lot @MarcoGorelli for letting me know.